spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetReaders.java

```diff
@@ -111,6 +124,49 @@ private SparkParquetReaders() {
     }
   }
 
+  private static class NameBasedReadBuilder extends ReadBuilder {
```

Comment:
I don't think a name-based reader is necessary. Instead, the file schema passed to the builder should have IDs assigned.

Reply:
I 'm OK to remove this to make the parquet reader consistent.